ts/Dashboards/Components/KPIComponent.ts

```diff
@@ -332,6 +332,7 @@ class KPIComponent extends Component {
 
 
     public render(): this {
+        this.emit({ type: 'beforeRender' });
```

Comment:
Not strictly related, but that is something, I already wondered in other components. We have `load` / `afterLoad`, `update` / `afterUpdate`, etc.; but here it is `beforeRender` / `afterRender`. We should be consistent and I think the best way to address this is actually to move the event as `render` into the base component class, so it gets emitted with the super call below like with the rest of events.
Any thoughts? @Vazonik , @sebastianbochan, @karolkolodziej , @pawellysy , @goransle

Reply:
I agree :)